src/Mobs/Monster.h

```diff
@@ -203,8 +208,13 @@ class cMonster :
 	/** Finishes a pathfinding task, be it due to failure or something else */
 	inline void FinishPathFinding(void)
 	{
-		m_TraversedCoordinates.clear();
 		m_bMovingToDestination = false;
+		if (m_Path != nullptr)
+		{
+			delete m_Path;
+			m_Path = nullptr;
+			
+		}
```

Comment:
I probably misunderstood you,  I don't see what's the relationship between a large function and moving it into CPP. (I assume CPP stands for cPlusPlus?)

I plan to migrate everything to smart pointers pretty soon.